### PR TITLE
Add ToAttachment method for OAuthCard

### LINKED ISSUE #4040: Add ToAttachment() method to OAuthCard
See [parent](https://github.com/microsoft/botframework-sdk/issues/5871)

### PR DESCRIPTION
**Pull request by ceciliaavila**
Fixes #4040

## Description
- Add the **_ToAttachment_** method for OAuthCard in _CardEx_ class.
- Add unit tests for all the ToAttachment methods.

## Testing
In the image below we can see the [CardsBot](https://github.com/microsoft/BotBuilder-Samples/tree/master/samples/csharp_dotnetcore/06.using-cards) sample using an OAuthCard as an attachment.

image

### CODE REVIEW

File: tests/Microsoft.Bot.Schema.Tests/Microsoft.Bot.Schema.Tests.csproj [hunk 1]
```diff
@@ -13,6 +13,7 @@
 	</ItemGroup>
 
 	<ItemGroup>
+		<ProjectReference Include="..\..\libraries\Microsoft.Bot.Builder\Microsoft.Bot.Builder.csproj" />
```
Comment (gabog) [inline tests/Microsoft.Bot.Schema.Tests/Microsoft.Bot.Schema.Tests.csproj:16]: Q: Why do we need to reference BotBuilder in this project?

Comment (ceciliaavila) [reply tests/Microsoft.Bot.Schema.Tests/Microsoft.Bot.Schema.Tests.csproj:16]: We added it so we can use the MessageFactory class, but we can refactor the test to remove it.

Comment (gabog) [reply tests/Microsoft.Bot.Schema.Tests/Microsoft.Bot.Schema.Tests.csproj:16]: That would be good so we don't depend on that project at the Schema level.

Thanks

Comment (ceciliaavila) [reply tests/Microsoft.Bot.Schema.Tests/Microsoft.Bot.Schema.Tests.csproj:16]: Ok, we'll have this for tomorrow. Thanks!

File: tests/Microsoft.Bot.Schema.Tests/CardExTest.cs [hunk 2]
```diff
@@ -0,0 +1,259 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Collections.Generic;
+using Microsoft.Bot.Builder;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+
```
Comment (mrivera-ms) [inline tests/Microsoft.Bot.Schema.Tests/CardExTest.cs:7]: There is a parallel effort that is migrating all existing tests to XUnit. Let's not add new MSTests. Instead, please create these using XUnit.

Comment (ceciliaavila) [reply tests/Microsoft.Bot.Schema.Tests/CardExTest.cs:6]: Changes applied. Thanks!

File: tests/Microsoft.Bot.Schema.Tests/Microsoft.Bot.Schema.Tests.csproj (hunk 1 above)
Comment (ceciliaavila) [reply tests/Microsoft.Bot.Schema.Tests/Microsoft.Bot.Schema.Tests.csproj]: Changes applied. Thanks!

**Pull request merged**